samples/extensions/open_gl_interop/open_gl_interop.cpp

```diff
@@ -0,0 +1,931 @@
+		    EGL_RED_SIZE, 8,
+		    EGL_GREEN_SIZE, 8,
+		    EGL_BLUE_SIZE, 8,
+		    EGL_ALPHA_SIZE, 8,
+		    EGL_NONE};
+
+		EGLint numConfigs;
+		eglChooseConfig(eglDisp, confAttr, &eglConf, 1, &numConfigs);
+
+		// Create a EGL context
+		constexpr EGLint ctxAttr[] = {EGL_CONTEXT_CLIENT_VERSION, 2, EGL_NONE};
+
+		eglCtx = eglCreateContext(eglDisp, eglConf, EGL_NO_CONTEXT, ctxAttr);
+
+		// Create an offscreen pbuffer surface, and then make it current
+		constexpr EGLint surfaceAttr[] = {EGL_WIDTH, 10, EGL_HEIGHT, 10, EGL_NONE};
+		eglSurface                     = eglCreatePbufferSurface(eglDisp, eglConf, surfaceAttr);
+		eglMakeCurrent(eglDisp, eglSurface, eglSurface, eglCtx);
+		gladLoadGLES2Loader((GLADloadproc) &eglGetProcAddress);
+#else
+		glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_API);
+		glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
+		glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 6);
+		glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+		glfwWindowHint(GLFW_OPENGL_DEBUG_CONTEXT, 1);
+		glfwWindowHint(GLFW_VISIBLE, GLFW_FALSE);
+		window = glfwCreateWindow(SHARED_TEXTURE_DIMENSION, SHARED_TEXTURE_DIMENSION, "OpenGL Window", nullptr, nullptr);
+		glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
+		glfwMakeContextCurrent(window);
+		gladLoadGL();
+#endif
+
+		glDebugMessageCallback(debugMessageCallback, NULL);
+		glEnable(GL_DEBUG_OUTPUT_SYNCHRONOUS);
+
+		program   = buildProgram(OPENG_VERTEX_SHADER, OPENGL_FRAGMENT_SHADER);
+		startTime = getTime();
+
+		glDisable(GL_DEPTH_TEST);
+
+		// Create the texture for the FBO color attachment.
+		// This only reserves the ID, it doesn't allocate memory
+		glGenTextures(1, &color);
+		glBindTexture(GL_TEXTURE_2D, color);
+
+		// Create the GL identifiers
+
+		// semaphores
+		glGenSemaphoresEXT(1, &glReady);
+		glGenSemaphoresEXT(1, &glComplete);
+		// memory
+		glCreateMemoryObjectsEXT(1, &mem);
+
+		// Platform specific import.
+		glImportSemaphore(glReady, GL_HANDLE_TYPE, handles.glReady);
+		glImportSemaphore(glComplete, GL_HANDLE_TYPE, handles.glComplete);
+		glImportMemory(mem, memorySize, GL_HANDLE_TYPE, handles.memory);
+
+		// Use the imported memory as backing for the OpenGL texture.  The internalFormat, dimensions
+		// and mip count should match the ones used by Vulkan to create the image and determine it's memory
+		// allocation.
+		glTextureStorageMem2DEXT(color, 1, GL_RGBA8, SHARED_TEXTURE_DIMENSION, SHARED_TEXTURE_DIMENSION, mem, 0);
+		glBindTexture(GL_TEXTURE_2D, 0);
+
+		// The remaining initialization code is all standard OpenGL
+		glGenVertexArrays(1, &vao);
+		glBindVertexArray(vao);
+
+		glGenFramebuffers(1, &fbo);
+		glBindFramebuffer(GL_FRAMEBUFFER, fbo);
+		glFramebufferTexture(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, color, 0);
+
+		glUseProgram(program);
+		glProgramUniform3f(program, 0, (float) SHARED_TEXTURE_DIMENSION, (float) SHARED_TEXTURE_DIMENSION, 0.0f);
+
+		glViewport(0, 0, SHARED_TEXTURE_DIMENSION, SHARED_TEXTURE_DIMENSION);
+	}
+
+	void render()
```

Comment:
I was confused between `OpenGLWindow::render()`, `OpenGLInterop::render(float)` and `draw()`. Could all these be concatenated together in `OpenGLInterop::render(float)`?

It may result in a bigger function but would be less confusing